automation_tools/__init__.py

```diff
@@ -1467,3 +1471,249 @@ def setenforce(mode):
+
+
+def delete_openstack_instance(instance_name):
+    """Deletes openstack Instance.
+
+    :param instance_name: A string. Openstack instance name to delete.
+
+    The following environment variables affect this command:
+
+    USERNAME:
+        The username of an openstack project to login.
+    PASSWORD:
+        The password of an openstack project to login.
+    AUTH_URL:
+        The authentication url of the project.
+    PROJECT_ID:
+        Project ID of an openstack project.
+
+    """
+    openstack_client = get_OpenstackClient()
+    try:
+        instance = openstack_client.servers.find(name=instance_name)
+    except novaclient.exceptions.NotFound:
+        print('Instance {0} not found in Openstack project.'.format(
+            instance_name
+        ))
+        return
+    instance.delete()
+
+
+def satellite6_upgrade(admin_password=None):
+    """Upgrades satellite from already created Openstack image
+    of old Satellite version to latest Satellite version compose.
+
+    The following environment variables affect this command:
+
+    ADMIN_PASSWORD:
+        Optional, defaults to 'changeme'. Foreman admin password.
+    BASE_URL:
+        URL for the compose repository.
+
+    """
+    base_url = os.environ.get('BASE_URL')
+    if base_url is None:
+        print('The BASE_URL environment variable should be defined')
+        sys.exit(1)
+    if admin_password is None:
+        admin_password = os.environ.get('ADMIN_PASSWORD', 'changeme')
+    # Removing rhel-released and rhel-optional repo
+    run('rm -rf /etc/yum.repos.d/rhel-{optional,released}.repo')
+    # Update the packages
+    update_packages(quiet=True)
+    # Setting Satellite61 Repos
+    repos = [
+        'rhel-{0}-server-satellite-6.1-rpms',
+        'rhel-{0}-server-satellite-6.0-rpms'
+    ]
+    enable_repos(repos[0].format(distro_info()[1]))
+    disable_repos(repos[1].format(distro_info()[1]))
+    # Add Sat6 repo from latest compose
+    satellite_repo = StringIO()
+    satellite_repo.write('[sat6]\n')
+    satellite_repo.write('name=satellite 6\n')
+    satellite_repo.write('baseurl={0}\n'.format(base_url))
+    satellite_repo.write('enabled=1\n')
+    satellite_repo.write('gpgcheck=0\n')
+    put(local_path=satellite_repo, remote_path='/etc/yum.repos.d/sat6.repo')
+    satellite_repo.close()
+    # Stop katello services, except mongod
+    run('katello-service stop')
+    run('service-wait mongod start')
+    # yum cleaning all
+    run('yum clean all', warn_only=True)
+    # Updating the packages again after setting sat6 repo
+    update_packages(quiet=True)
+    # Upgrading Katello installer
+    run('katello-installer --upgrade')
+    # Test the Upgrade is successful
+    run('hammer -u admin -p {0} ping'.format(admin_password))
+
+
+def product_upgrade(product, instance_name, image_name, flavor_name, ssh_key):
+    """Task which upgrades the product.
+
+    Products are satellite, capsule or client.
```

Comment:
As of now , I have implemented only for satellite. For capsule and client soon the functions will there.
Removing capsule and client.